test/e2e/certrotation.go

```diff
@@ -0,0 +1,157 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/clock"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	testlibrary "github.com/openshift/cluster-kube-apiserver-operator/test/library"
+	configv1helpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	g "github.com/onsi/ginkgo/v2"
+)
+
+var _ = g.Describe("[sig-api-machinery] kube-apiserver operator", func() {
+	g.It("[Operator][Serial] TestCertRotationTimeUpgradeable", func() {
+		testCertRotationTimeUpgradeable(g.GinkgoTB())
+	})
+	g.It("[Operator][Serial] TestCertRotationStompOnBadType", func() {
+		testCertRotationStompOnBadType(g.GinkgoTB())
+	})
+})
+
+func testCertRotationTimeUpgradeable(t testing.TB) {
+	kubeConfig, err := testlibrary.NewClientConfigForTest()
```

Comment:
could you adda a new tmp commit with t.Fail() just to see if the expected CI jobs will fail?